docs/data-companion/pruning.md

```diff
@@ -0,0 +1,338 @@
+```
+
+## Retain Height
+
+One important concept that can affect the pruning of nodes is the `retain height`. The retain height determines the specific
+height from which the data can be safely deleted from the node's storage. By considering the retain height,
+nodes can effectively manage their storage usage and ensure that they are only retaining the data that is necessary for
+their operations. This is important because storage space is a finite resource and nodes with limited storage space may
+struggle to keep up with the growth of the blockchain.
+
+## Pruning Blocks
+
+The pruning service uses the "block retain height" parameter to specify the height to which the node will
+preserve blocks. It is important to note that this parameter differs from the application block retain height, which
+the application sets in response to ABCI commit messages.
+
+> NOTE: In order to set the block retain height on the node, you have to enable the privileged services endpoint and the
+pruning service in the configuration as described in section above.
+
+Once the services are enabled, you can use the Golang client provided by CometBFT to invoke the method that sets the block retain height.
+
+Here is an example code:
+```
+
+import (
+    "github.com/cometbft/cometbft/rpc/grpc/client/privileged"
+)
+
+ctx := context.Background()
+
+// Privileged Service Client
+addr := "0.0.0.0:26091"
+conn, err := privileged.New(ctx, addr, privileged.WithInsecure())
+if err != nil {
+    // Do something with the error
+}
+
+err := conn.SetBlockRetainHeight(ctx, height)
+if err != nil {
+    // Do something with the error
+}
+```
+
+If you need to check what is the current value for the `Block Retain Height` you can use another method.
+
+Here's an example:
+```
+retainHeight, err := conn.GetBlockRetainHeight(ctx)
+if err != nil {
+    // Do something with the error
+} else {
+    // Do something with
+    // `retainHeight.App`
+    // `retainHeight.PruningService`
+}
+```
+
+Retaining data is crucial to data management, and the application has complete control over the application retain height.
+The operator can monitor the application retain height with `GetBlockRetainHeight`, which returns a `RetainHeights`
+structure with both the block retain height and the application retain height (as shown in the code above).
+
+It's worth noting that at any given point in time, the node will only accept the lowest retained height.
+If you try to set the `Block Retain Height` to a value that is lower to what is currently stored in the node, an error will
+be returned informing that.
+
+By default, both the application retain height and the data companion retain height are set to zero. This is done to prevent
+either one of them from prematurely pruning the data while the other has not indicated that it's okay to do so.
+
+In essence, the node will preserve blocks up to the lowest value between data companion block retain height and the application
```

Comment:
Ah I just saw this so disregard my comment above. But I would move this section up front, before the example. It is flows more natural in my opinion (but it is not a strong preference).

Reply:
OK, I've moved the section up before the example, kept the original text (not the suggested one above since they are the same)